src/components/Admin/TrustedSources.js

```diff
@@ -17,24 +18,43 @@ class TrustedSources extends React.Component {
   }
 
   componentDidMount() {
-    this.props.flux.actions.ADMIN.load_trusted_sources(PIPELINE_KEYS);
+    this.props.flux.actions.ADMIN.notifyDataGridTrustedSourcesLoaded();
   }
 
   handleSave(rows) {
-    rows.map(row => this.appendRowKey(row));
+    rows.map(row => row.rowKey = this.getRowKey(row));
     this.props.flux.actions.ADMIN.save_trusted_sources(rows);
   }
 
-  appendRowKey(row) {
+  getRowKey(row) {
     return row.pipelinekey + ',' + row.externalsourceid + ',' + row.sourcetype + ',' + row.rank;
   }
 
-  handleRemove(rows) {
-    this.props.flux.actions.ADMIN.remove_trusted_sources(rows);
+  handleRemove(rows) {    
+    const sourcesWithAllFieldsSet = this.filterSourcesWithUnsetFields(rows);
+    if (this.trustedSourcesToRemoveExist(sourcesWithAllFieldsSet)) {
+      this.props.flux.actions.ADMIN.remove_trusted_sources(sourcesWithAllFieldsSet);
+    }
+  }
+
+  filterSourcesWithUnsetFields(sources) {
+    return sources.filter(source => source.pipelinekey.length > 0 || source.externalsourceid.length > 0 || source.sourcetype.length > 0 || source.rank.length > 0)
+  }
+
+  trustedSourcesToRemoveExist(sources) {
+    const shouldRemove = sources.length > 0;
+    return shouldRemove;
   }
 
   getTrustedSourcesColumns() {
+    const pipelineKeys = [
```

Comment:
I'm confused. Shouldn't the enabled pipelines for the site come from the graphql service instead of being hard-coded? There's a property on the `DataStore` that has a list of all the sources enabled for the current site.

Reply:
ok yea that's correct, I'll change that.